src/content/en/fundamentals/instant-and-offline/pwa-architecture/index.md

```diff
@@ -0,0 +1,922 @@
+
+Service worker caching should be considered a progressive enhancement. If your web app follows the model of conditionally registering a service worker only if it is supported, then you get offline support on browsers with service workers and on browsers that do not support service workers. The offline-specific code is never called and there is no overhead or breakage for older browsers. 
+
+
+### Key Concepts {: #key-concepts }
+
+The app shell approach relies on caching the "shell" of your web application using a service worker. Using the **_app shell + dynamic content model_** greatly improves app performance and works really well with service worker caching as a progressive enhancement. Progressively enhancing your web app means you can gradually add in features like offline caching, push notifications, and add-to-home-screen. 
+
+Here is a high-level description of how it works:
+
+1. When the user accesses your website the basic HTML, JavaScript, and CSS display. On the initial website visit, the page registers the service worker that controls future navigations on the site. Registration creates a new service worker instance and triggers the install event that the service worker responds to. Also, the app shell content is added to the cache. Once installed, the service worker controls future navigations on the site.
+
+2. After the shell content loads, the app requests content to populate the view. The app shell plus dynamic content equals the compete rendered page. Next, the SPA requests content (for example, via [XMLHttpRequest](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest) or the [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API)) and page content is fetched and used to populate the view. And, once those handlers complete, your service worker enters into an idle state. So, the service worker is running but it is not doing anything until a network request fires off a new event. And, in response to a network request, a fetch event handler intercepts the request and responds as you see fit. After a period of idleness your service worker script is stopped automatically but when the next network request is made when the page is loaded again the service worker is started back up and can immediately respond to fetch events. 
+
+<img src="images/0.png" alt="After a period of idleness your service worker script is stopped automatically but when the next network request is made when the page is loaded again the service worker is started back up and can immediately respond to fetch events." class="screenshot"/>
```

Comment:
Based on the description you have, I'd recommend wrapping this in a figure and using the figcaption tag in addition to the alt tag.